### ISSUE TITLE
separate this module into core and a gulp wrapper

### ISSUE THREAD

**Issue by stevemao** [caitp/gulp-ddescribe-iit#7]
same as https://github.com/btford/grunt-ddescribe-iit/issues/14

**Comment by caitp** [comment]
there's no actual gulp code in this other than gulp-util's error class (for convention). this works fine with any stream-based abstraction

**Comment by stevemao** [comment]
Well if this is actually a vanilla node module it shouldn't be a gulp plugin. According to the [guideline](https://github.com/gulpjs/gulp/blob/master/docs/writing-a-plugin/guidelines.md#guidelines) 

> Name your plugin appropriately: it should begin "gulp-" if it is a gulp plugin
> If it is not a gulp plugin, it should not begin "gulp-"

**Comment by stevemao** [comment]
https://github.com/caitp/gulp-ddescribe-iit/blob/master/index.js#L207-L209

``` js
return through2.obj(function processFile(file, enc, cb) {
    if (file.isNull()) return cb(null, file);
    if (file.isStream()) return cb(new PluginError('gulp-ddescribe-iit', 'Streaming not supported'));
```

This is actually using vinyl. Unless your stream line is also using vinyl then it works fine with none gulp plugins. But usually if a stream module using vinyl its a gulp plugin

**Comment by caitp** [comment]
you can push that interface into any stream, there is no requirement on actually using Vinyl, and even if there were, it's a fairly widely used api.

But more importantly, it's a small library with an unenforceable license, why not just fork it and build whatever adaptation you need?

**Comment by stevemao** [comment]
True. I just thought a core-wrapper system would work better. Including everything in the module is a slippery slope. What about Broccoli? Or the next build tools? People have to rewrite the core logic in the plugins. The beauty of the node ecosystem is small focused modules. Since there are similar logic in btford/grunt-ddescribe-iit, It's kinda reinventing the wheel. 
I'm just proposing to merge the core logics into a core module, then plugins can just use it.

Just my 2c :)

**Comment by caitp** [comment]
I'm just saying, the library is small enough and liberally licensed enough that it really doesn't matter, if someone needs it used differently, they can create a version that works for them. I've already created a different version of this one just to work around the false-positives reported by this one (and Brian's grunt plugin)
